Change SSH executables copy to Docker COPY and standardize app root

andreriesco opened the pull request:
. Changes the copy of executables from the SSH on the running
container to copy them with COPY in the Dockerfiles on the container
build.
. Standardizes between the SDK, the debug and the production the
APP_ROOT directory variable, to use a value defined on settings.json,
making it clear for the user where his project files are.
Does the same for the APP_EXECUTABLE file.
. Adds the information about how to deal with extra files (not added to
the running container automatically) on the documentation of each
template.

microhobby commented on cmakeConsole/.vscode/tasks.json:
```diff
@@ -35,7 +35,7 @@
                 "-DCMAKE_CXX_COMPILER=arm-linux-gnueabihf-g++",
                 "-DCMAKE_C_COMPILER=arm-linux-gnueabihf-gcc",
-                "-Bbuild-armhf"
+                "-Bbuild-arm"
```
so, this is wrong also on the `bookworm` branch?

microhobby commented on cmakeConsole/.vscode/tasks.json:
```diff
@@ -35,7 +35,7 @@
                 "-DCMAKE_CXX_COMPILER=arm-linux-gnueabihf-g++",
                 "-DCMAKE_C_COMPILER=arm-linux-gnueabihf-gcc",
-                "-Bbuild-armhf"
+                "-Bbuild-arm"
```
so, this is wrong also on the `bookworm` branch?

andreriesco replied:
It is not wrong actually, it is just that if we use `build-arm` instead of `build-armhf` we can pass the build directory like this on the `COPY build-${IMAGE_ARCH}/debug ${APP_ROOT}/app`

Commit:
cpp: Change SSH executables copy to Docker COPY and standardize app root

This commit:
. Changes the copy of executables from the SSH on the running
container to copy them with COPY in the Dockerfiles on the container
build.
. Standardizes between the SDK, the debug and the production the
APP_ROOT directory variable, to use a value defined on settings.json,
making it clear for the user where his project files are.
Does the same for the APP_EXECUTABLE file.
. Adds the information about how to deal with extra files (not added to
the running container automatically) on the documentation of each
template.

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

Commit:
cmakeConsole: Remove Riscv and amd64 debug tasks at launch.json

Remove Riscv and amd64 debug tasks at launch.json, as there are no
tasks on tasks.json implemented on this template for this
architectures

Signed-off-by: Andre Riesco <andre.riesco@toradex.com>

microhobby commented:
LGTM
